File: app/src/main/java/com/worldventures/dreamtrips/core/session/CirclesInteractor.java
package com.worldventures.dreamtrips.core.session;

import com.worldventures.dreamtrips.core.janet.SessionActionPipeCreator;
import com.worldventures.dreamtrips.modules.common.api.janet.command.GetCirclesCommand;

import io.techery.janet.ActionPipe;
import rx.schedulers.Schedulers;

public class CirclesInteractor {

   ActionPipe<GetCirclesCommand> pipe;

   public CirclesInteractor(SessionActionPipeCreator sessionActionPipeCreator) {
      pipe = sessionActionPipeCreator.createPipe(GetCirclesCommand.class, Schedulers.io());
   }

   public ActionPipe<GetCirclesCommand> pipe() {
      return pipe;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/users/base/interactor/CirclesInteractor.kt
package com.worldventures.dreamtrips.social.service.users.base.interactor

import com.worldventures.core.janet.SessionActionPipeCreator
import com.worldventures.dreamtrips.social.service.users.circle.command.GetCirclesCommand

import io.techery.janet.ActionPipe
import rx.schedulers.Schedulers

open class CirclesInteractor(pipeCreator: SessionActionPipeCreator) {
   val pipe: ActionPipe<GetCirclesCommand> = pipeCreator.createPipe(GetCirclesCommand::class.java, Schedulers.io())
}
